src/CommunityToolkit/Xamarin.CommunityToolkit/Extensions/Navigation/NavigationExtensions.shared.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Threading.Tasks;
+using Xamarin.CommunityToolkit.UI.Views;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Extensions
+{
+	/// <summary>
+	/// Extension methods for <see cref="INavigation"/>.
+	/// </summary>
+	public static partial class NavigationExtensions
```

Comment:
I think it would make sense to add PageExtensions with the same methods.

this.ShowPopup() is betteer than this.Navigation.ShowPopup()

Reply:
Are you thinking in addition or have both sets of extension methods?

Reply:
I think that using this API from Page-level is more convenient, but, of course, we can preserve both sets of methods (Page methods will call Navigation methods).

Reply:
Makes sense to me, I'll add it in today. 

It is important to have it in the INavigation object for building a NavigationService in MVVM libraries. Typically they have direct access to this object.

Reply:
I added the extension method for `NavigableElement` as it is the highest parent up the inheritance tree that contains `INavigation`. The extension methods I implemented in the latest code push just wrap the existing extension methods to keep it to 1 code path. It made sense to give this API the highest level of visibility across the API.

```c#
public static class NavigableElementExtensions
{
    public static void ShowPopup(this NavigableElement element, BasePopup popup) =>
        element.Navigation.ShowPopup(popup);

    public static Task<T> ShowPopupAsync<T>(this NavigableElement element, Popup<T> popup) =>
        element.Navigation.ShowPopupAsync(popup);
}
```